### ISSUE TITLE
Ingester option replication_factor=2 is broken

### ISSUE THREAD

**Issue by MalKeshar** [grafana/mimir#4993]
Ingester option replication_factor=2 is broken. We have 3 instances of ingester. Stopping one them leads to broken write and data loss. Distributor shows error: "at least 2 live replicas required across different availability zones, could only find 1 - unhealthy instances" in logs.

Case the same as in https://github.com/grafana/mimir/discussions/4005, but unregister_on_shutdown disabled and our hash ring for ingesters is fine. All three ingesters joined it.
Most important config values:
ingester.ring.replication_factor: 2
unregister_on_shutdown: false (default)
ingester.ring.zone_awareness_enabled: true (default)

Where is no information in docs
https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/ingester/
https://github.com/grafana/mimir/blob/main/docs/sources/mimir/references/architecture/components/ingester.md
that replication_factor = 2 should not be used
 
Would be great to have ability to use it, because it allows to reduce overall load on the cluster. If it not possible, please, mention this fact in documentation.

Details
* Mimir version: 2.8.0
* Helm chart version: 4.4.1
* Changed config directives:
[https://gist.github.com/MalKeshar/b4a20e9ca76552d8bc61d7210ffcb704](https://gist.github.com/MalKeshar/b4a20e9ca76552d8bc61d7210ffcb704)
* Ingester hash ring status:
![Screenshot from 2023-05-12 18-07-40](https://github.com/grafana/mimir/assets/13730780/347fd21e-f8dc-48f2-802a-216aca9b27e6)
* Distributor log error:
```
ts=2023-05-12T15:06:47.099598957Z caller=pool.go:193 level=warn msg="removing ingester failing healthcheck" addr=172.25.237.90:9095 reason="rpc error: code = DeadlineExceeded desc = context deadline exceeded"
ts=2023-05-12T15:06:47.527288369Z caller=logging.go:86 level=warn traceID=5d5eb73327a2c959 msg="POST /api/v1/push (500) 614.965µs Response: \"at least 2 live replicas required across different availability zones, could only find 1 - unhealthy instances: 172.25.237.90:9095\\n\" ws: false; Connection: close; Content-Encoding: snappy; Content-Length: 21807; Content-Type: application/x-protobuf; User-Agent:
```

**Comment by jhalterman** [comment]
@MalKeshar You're right that the docs are unclear on this. They're being updated via #4934 to better explain how replication effects availability. 

Mimir requires that a majority of ingesters be available. This means that with RF=3 we can lose 1 ingester since we'd still have a majority available, but with RF=2 or RF=1 we cannot lose any ingesters. This highlights how running with RF=2 provides no HA benefit over running with RF=1. So overall, RF=3 is recommended for production.

**Comment by pracucci** [comment]
> i've seen the ring go down if i have 40 ingesters and 2 go offline with OOM if you have RF=3

That's right. When you run Mimir without zone-aware replication, the required quorum is "all ingesters minus 1". When you run Mimir with zone-aware replication, the required quorum is ingesters in "2 out of 3 zones" (assuming RF=3 and zones=3).
